compile error in linux x86 gcc5.5 [12545]

/data/workspace/Fast-DDS/src/cpp/rtps/flowcontrol/FlowControllerFactory.cpp:24:87:   required from here
**/usr/include/c++/5/ext/new_allocator.h:120:4: error: use of deleted function ‘constexpr std::pair<_T1, _T2>::pair(const std::pair<_T1, _T2>&) [with _T1 = const std::__cxx11::basic_string<char>; _T2 = std::unique_ptr<eprosima::fastdds::rtps::FlowController>]’**
  { ::new((void *)__p) _Up(std::forward<_Args>(__args)...); }
    ^
In file included from /usr/include/c++/5/bits/stl_algobase.h:64:0,
                 from /usr/include/c++/5/bits/char_traits.h:39,
                 from /usr/include/c++/5/ios:40,
                 from /usr/include/c++/5/ostream:38,
                 from /usr/include/c++/5/iostream:39,
                 from /data/workspace/Fast-DDS/include/fastdds/rtps/common/Time_t.h:24,
                 from /data/workspace/Fast-DDS/include/fastdds/rtps/attributes/WriterAttributes.h:22,
                 from /data/workspace/Fast-DDS/src/cpp/rtps/flowcontrol/FlowControllerFactory.hpp:5,
                 from /data/workspace/Fast-DDS/src/cpp/rtps/flowcontrol/FlowControllerFactory.cpp:1:
/usr/include/c++/5/bits/stl_pair.h:127:17: note: ‘constexpr std::pair<_T1, _T2>::pair(const std::pair<_T1, _T2>&) [with _T1 = const std::__cxx11::basic_string<char>; _T2 = std::unique_ptr<eprosima::fastdds::rtps::FlowController>]’ is implicitly deleted because the default definition would be ill-formed:
       constexpr pair(const pair&) = default;
                 ^
**/usr/include/c++/5/bits/stl_pair.h:127:17: error: use of deleted function ‘std::unique_ptr<_Tp, _Dp>::unique_ptr(const std::unique_ptr<_Tp, _Dp>&) [with _Tp = eprosima::fastdds::rtps::FlowController; _Dp = std::default_delete<eprosima::fastdds::rtps::FlowController>]’**

--- a/src/cpp/rtps/flowcontrol/FlowControllerFactory.cpp
+++ b/src/cpp/rtps/flowcontrol/FlowControllerFactory.cpp
@@ -21,20 +21,23 @@ void FlowControllerFactory::init(
     // Create default flow controllers.
 
     // PureSyncFlowController -> used by volatile besteffort writers.
-    flow_controllers_.insert({pure_sync_flow_controller_name,
-                              std::unique_ptr<FlowController>(
-                                  new FlowControllerImpl<FlowControllerPureSyncPublishMode,
-                                  FlowControllerFifoSchedule>(participant_, nullptr))});
+    flow_controllers_.insert(std::make_pair<std::string, std::unique_ptr<FlowController>>(
+                pure_sync_flow_controller_name,
+                std::unique_ptr<FlowController>(
+                    new FlowControllerImpl<FlowControllerPureSyncPublishMode,
+                    FlowControllerFifoSchedule>(participant_, nullptr))));
     // SyncFlowController -> used by rest of besteffort writers.
-    flow_controllers_.insert({sync_flow_controller_name,
-                              std::unique_ptr<FlowController>(
-                                  new FlowControllerImpl<FlowControllerSyncPublishMode,
-                                  FlowControllerFifoSchedule>(participant_, nullptr))});
+    flow_controllers_.insert(std::make_pair<std::string, std::unique_ptr<FlowController>>(
+                sync_flow_controller_name,
+                std::unique_ptr<FlowController>(
+                    new FlowControllerImpl<FlowControllerSyncPublishMode,
+                    FlowControllerFifoSchedule>(participant_, nullptr))));
     // AsyncFlowController
-    flow_controllers_.insert({async_flow_controller_name,
-                              std::unique_ptr<FlowController>(
-                                  new FlowControllerImpl<FlowControllerAsyncPublishMode,
-                                  FlowControllerFifoSchedule>(participant_, nullptr))});
+    flow_controllers_.insert(std::make_pair<std::string, std::unique_ptr<FlowController>>(
+                async_flow_controller_name,
+                std::unique_ptr<FlowController>(
+                    new FlowControllerImpl<FlowControllerAsyncPublishMode,
+                    FlowControllerFifoSchedule>(participant_, nullptr))));
 
 #ifdef FASTDDS_STATISTICS
     flow_controllers_.insert({async_statistics_flow_controller_name,

